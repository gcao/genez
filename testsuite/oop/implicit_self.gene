#!/usr/bin/env gene run

# Test: Implicit self reference with /field

(class Person
  (var /name)
  (var /age)
  
  (.fn greet []
    (print "Hello, I am" /name))
  
  (.fn birthday []
    (= /age (+ /age 1)))
  
  (.fn info []
    (do
      (print "Name:" /name)
      (print "Age:" /age))))

(class BankAccount
  [balance 0]
  [account_number ""]
  
  (.fn deposit [amount]
    (= /balance (+ /balance amount)))
  
  (.fn withdraw [amount]
    (if (>= /balance amount)
      (= /balance (- /balance amount))
    else
      (print "Insufficient funds")))
  
  (.fn check_balance []
    /balance))

# Nested field access
(class Address
  [street ""]
  [city ""]
  [zip ""]
  
  (.fn full_address []
    (+ /street ", " /city " " /zip)))

(print "Implicit self syntax parsed successfully")