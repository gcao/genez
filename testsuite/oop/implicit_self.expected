Implicit self syntax parsed successfully